controllers/datadogagent/controller_reconcile_v2.go

```diff
@@ -123,9 +124,15 @@ func (r *Reconciler) reconcileInstanceV2(ctx context.Context, logger logr.Logger
 			errs = append(errs, featErr)
 		}
 	}
+	if len(errs) > 0 {
+		return r.updateStatusIfNeededV2(logger, instance, newStatus, result, errors.NewAggregate(errs))
+	}
```

Comment:
is it intended to put this block here as well as line 133? if so then we don't need to append to `errs` on line 132, just check for the return value..

Reply:
it was to differentiate errors appending during the `ManageDependencies()`
and the `override.Dependencies()`

maybe it is indeed better to only keep it after `override.Dependencies()` so we have one bloc of errors for the dependencies

Reply:
thanks - after thinking about it, I will keep the current behavior, just not append to the existing `errs` for clarity